Startup crash

### Describe the bug

The crash isn't happening consistently but, when it happens is it on node startup

### Has this worked before in a previous version?

```text

```

### 🔬 Minimal Reproduction

```
participants_matrix:
  el:
    - el_type: geth
      el_image: ethpandaops/geth:prague-devnet-6
  cl:
    - cl_type: teku
      cl_image: consensys/teku:develop
      cl_extra_env_vars: {"JAVA_OPTS": "-Xmx3G -XX:UseSVE=0"}
    - cl_type: prysm
      cl_image: ethpandaops/prysm-beacon-chain:develop
      vc_image: ethpandaops/prysm-validator:develop
network_params:
  num_validator_keys_per_node: 32
  electra_fork_epoch: 100
additional_services:
  - dora
snooper_enabled: true
```

### Error

```text
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x35dac40]

goroutine 4652 [running]:
github.com/prysmaticlabs/prysm/v5/beacon-chain/forkchoice/doubly-linked-tree.(*Store).setOptimisticToInvalid(0x400015c960, {0x1b70970, 0x4001edf0e0}, {0xca, 0xe0, 0x56, 0xb6, 0xa3, 0x4c, 0x88, ...}, ...)
        beacon-chain/forkchoice/doubly-linked-tree/optimistic_sync.go:26 +0x50
github.com/prysmaticlabs/prysm/v5/beacon-chain/forkchoice/doubly-linked-tree.(*ForkChoice).SetOptimisticToInvalid(0x4001468118?, {0x1b70970?, 0x4001edf0e0?}, {0xca, 0xe0, 0x56, 0xb6, 0xa3, 0x4c, 0x88, ...}, ...)
        beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go:363 +0x5c
github.com/prysmaticlabs/prysm/v5/beacon-chain/blockchain.(*Service).notifyForkchoiceUpdate(0x400030a400, {0x1b70970?, 0x4001edf0e0?}, 0x4003c979e0)
        beacon-chain/blockchain/execution_engine.go:93 +0xc10
github.com/prysmaticlabs/prysm/v5/beacon-chain/blockchain.(*Service).lateBlockTasks(0x400030a400, {0x1b70970, 0x4001edf0e0})
        beacon-chain/blockchain/process_block.go:678 +0x9a0
github.com/prysmaticlabs/prysm/v5/beacon-chain/blockchain.(*Service).runLateBlockTasks(0x400030a400)
        beacon-chain/blockchain/process_block.go:494 +0x184
created by github.com/prysmaticlabs/prysm/v5/beacon-chain/blockchain.(*Service).Start in goroutine 4939
        beacon-chain/blockchain/service.go:217 +0x13c
```

### Platform(s)

_No response_

### What version of Prysm are you running? (Which release)

_No response_

### Anything else relevant (validator index / public key)?

_No response_

--- a/beacon-chain/blockchain/chain_info_forkchoice.go
+++ b/beacon-chain/blockchain/chain_info_forkchoice.go
@@ -3,10 +3,13 @@ package blockchain
 import (
 	"context"
 
+	"github.com/pkg/errors"
 	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state"
 	consensus_blocks "github.com/prysmaticlabs/prysm/v5/consensus-types/blocks"
 	"github.com/prysmaticlabs/prysm/v5/consensus-types/forkchoice"
 	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v5/encoding/bytesutil"
+	"github.com/prysmaticlabs/prysm/v5/runtime/version"
 )
 
 // CachedHeadRoot returns the corresponding value from Forkchoice
@@ -99,4 +102,27 @@ func (s *Service) ParentRoot(root [32]byte) ([32]byte, error) {
 	s.cfg.ForkChoiceStore.RLock()
 	defer s.cfg.ForkChoiceStore.RUnlock()
 	return s.cfg.ForkChoiceStore.ParentRoot(root)
+}
+
+// hashForGenesisBlock returns the right hash for the genesis block
+func (s *Service) hashForGenesisBlock(ctx context.Context, root [32]byte) ([]byte, error) {
+	genRoot, err := s.cfg.BeaconDB.GenesisBlockRoot(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get genesis block root")
+	}
+	if root != genRoot {
+		return nil, errNotGenesisRoot
+	}
+	st, err := s.cfg.BeaconDB.GenesisState(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get genesis state")
+	}
+	if st.Version() < version.Bellatrix {
+		return nil, nil
+	}
+	header, err := st.LatestExecutionPayloadHeader()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get latest execution payload header")
+	}
+	return bytesutil.SafeCopyBytes(header.BlockHash()), nil
 }

--- a/beacon-chain/blockchain/chain_info_test.go
+++ b/beacon-chain/blockchain/chain_info_test.go
@@ -613,3 +613,20 @@ func TestService_IsFinalized(t *testing.T) {
 	require.Equal(t, true, c.IsFinalized(ctx, br))
 	require.Equal(t, false, c.IsFinalized(ctx, [32]byte{'c'}))
 }
+
+func Test_hashForGenesisRoot(t *testing.T) {
+	beaconDB := testDB.SetupDB(t)
+	ctx := context.Background()
+	c := setupBeaconChain(t, beaconDB)
+	st, _ := util.DeterministicGenesisStateElectra(t, 10)
+	require.NoError(t, c.cfg.BeaconDB.SaveGenesisData(ctx, st))
+	root, err := beaconDB.GenesisBlockRoot(ctx)
+	require.NoError(t, err)
+	genRoot, err := c.hashForGenesisBlock(ctx, [32]byte{'a'})
+	require.ErrorIs(t, err, errNotGenesisRoot)
+	require.IsNil(t, genRoot)
+
+	genRoot, err = c.hashForGenesisBlock(ctx, root)
+	require.NoError(t, err)
+	require.Equal(t, [32]byte{}, [32]byte(genRoot))
+}

--- a/beacon-chain/blockchain/error.go
+++ b/beacon-chain/blockchain/error.go
@@ -30,6 +30,8 @@ var (
 	ErrNotCheckpoint = errors.New("not a checkpoint in forkchoice")
 	// ErrNilHead is returned when no head is present in the blockchain service.
 	ErrNilHead = errors.New("nil head")
+	// errNotGenesisRoot is returned when the root is not the genesis block root.
+	errNotGenesisRoot = errors.New("root is not the genesis block root")
 )
 
 var errMaxBlobsExceeded = errors.New("Expected commitments in block exceeds MAX_BLOBS_PER_BLOCK")

--- a/beacon-chain/blockchain/execution_engine.go
+++ b/beacon-chain/blockchain/execution_engine.go
@@ -69,6 +69,18 @@ func (s *Service) notifyForkchoiceUpdate(ctx context.Context, arg *fcuConfig) (*
 		SafeBlockHash:      justifiedHash[:],
 		FinalizedBlockHash: finalizedHash[:],
 	}
+	if len(fcs.HeadBlockHash) != 32 || [32]byte(fcs.HeadBlockHash) == [32]byte{} {
+		// check if we are sending FCU at genesis
+		hash, err := s.hashForGenesisBlock(ctx, arg.headRoot)
+		if errors.Is(err, errNotGenesisRoot) {
+			log.Error("Sending nil head block hash to execution engine")
+			return nil, nil
+		}
+		if err != nil {
+			return nil, errors.Wrap(err, "could not get head block hash")
+		}
+		fcs.HeadBlockHash = hash
+	}
 	if arg.attributes == nil {
 		arg.attributes = payloadattribute.EmptyWithVersion(headBlk.Version())
 	}

--- a/beacon-chain/blockchain/execution_engine_test.go
+++ b/beacon-chain/blockchain/execution_engine_test.go
@@ -113,6 +113,7 @@ func Test_NotifyForkchoiceUpdate(t *testing.T) {
 	state, blkRoot, err = prepareForkchoiceState(ctx, 2, bellatrixBlkRoot, altairBlkRoot, params.BeaconConfig().ZeroHash, ojc, ofc)
 	require.NoError(t, err)
 	require.NoError(t, fcs.InsertNode(ctx, state, blkRoot))
+	badHash := [32]byte{'h'}
 
 	tests := []struct {
 		name             string
@@ -210,7 +211,7 @@ func Test_NotifyForkchoiceUpdate(t *testing.T) {
 			blk: func() interfaces.ReadOnlySignedBeaconBlock {
 				b, err := consensusblocks.NewSignedBeaconBlock(&ethpb.SignedBeaconBlockBellatrix{Block: &ethpb.BeaconBlockBellatrix{
 					Body: &ethpb.BeaconBlockBodyBellatrix{
-						ExecutionPayload: &v1.ExecutionPayload{},
+						ExecutionPayload: &v1.ExecutionPayload{BlockHash: badHash[:]},
 					},
 				}})
 				require.NoError(t, err)

--- a/beacon-chain/blockchain/execution_engine_test.go
+++ b/beacon-chain/blockchain/execution_engine_test.go
@@ -84,7 +84,7 @@ func Test_NotifyForkchoiceUpdate_GetPayloadAttrErrorCanContinue(t *testing.T) {
 	service.cfg.PayloadIDCache.Set(1, [32]byte{}, [8]byte{})
 	got, err := service.notifyForkchoiceUpdate(ctx, arg)
 	require.NoError(t, err)
-	require.DeepEqual(t, got, pid) // We still get a payload ID even though the state is bad. This means it returns until the end.
+	require.IsNil(t, got)
 }
 
 func Test_NotifyForkchoiceUpdate(t *testing.T) {

--- a/changelog/potuz_fix_genesis_proposal.md
+++ b/changelog/potuz_fix_genesis_proposal.md
@@ -1,0 +1,3 @@
+### Fixed
+
+- Fixed proposing at genesis when starting post Bellatrix

